src/ensembl/src/content/app/browser/zmenu/ZmenuContent.tsx

```diff
@@ -16,88 +16,105 @@
 
 import React from 'react';
 import classNames from 'classnames';
-import { connect } from 'react-redux';
 
-import { changeFocusObject } from 'src/content/app/browser/browserActions';
+import useGenomeBrowser from 'src/content/app/browser/hooks/useGenomeBrowser';
 
 import ZmenuAppLinks from './ZmenuAppLinks';
 
 import {
+  ZmenuContentGeneMetadata,
   ZmenuContentFeature as ZmenuContentFeatureType,
-  ZmenuContentLine as ZmenuContentLineType,
-  ZmenuContentBlock as ZmenuContentBlockType,
   ZmenuContentItem as ZmenuContentItemType,
+  ZmenuContentBlock as ZmenuContentBlockType,
   Markup
-} from './zmenu-types';
+} from 'ensembl-genome-browser';
 
 import styles from './Zmenu.scss';
 
 export type ZmenuContentProps = {
   content: ZmenuContentFeatureType[];
 };
 
-type ZmenuContentFeatureProps = {
-  id: string;
-  lines: ZmenuContentLineType[];
-};
-
-type ZmenuContentLineProps = {
-  blocks: ZmenuContentBlockType[];
-  id: string;
-};
+export const ZmenuContent = (props: ZmenuContentProps) => {
+  const { content } = props;
+
+  const { id } = content[1].metadata as ZmenuContentGeneMetadata;
+  const unversionedId = id.split('.')[0];
+  const featureId = `gene:${unversionedId}`;
+
+  const renderedContent = (
+    <>
+      {content.map((feature: any, index) => (
+        <p key={index} className={styles.zmenuContentFeature}>
+          <ZmenuContentFeature id={featureId} feature={feature} />
+        </p>
+      ))}
+      <ZmenuAppLinks featureId={featureId} />
+    </>
+  );
 
-type ZmenuContentBlockProps = {
-  items: ZmenuContentItemType[];
-  id: string;
+  return renderedContent;
 };
 
-export type ZmenuContentItemProps = ZmenuContentItemType & {
-  changeFocusObject: (objectId: string) => void;
+type ZmenuContentFeatureProps = {
   id: string;
+  feature: ZmenuContentFeatureType;
 };
-
-export const ZmenuContent = (props: ZmenuContentProps) => {
-  const features = props.content;
-  const renderedContent = features.map((feature) => (
-    <div key={feature.id}>
-      <ZmenuContentFeature id={feature.id} lines={feature.lines} />
-      <ZmenuAppLinks featureId={feature.id} />
-    </div>
-  ));
-  return <>{renderedContent}</>;
-};
-
 export const ZmenuContentFeature = (props: ZmenuContentFeatureProps) => {
+  let lineIndex = 0;
+
+  const linesAndBlocks: [ZmenuContentBlockType[]] = [[]];
+
+  props.feature.data.map((block) => {
+    if (block.type === 'line-break') {
+      lineIndex++;
+    } else {
+      if (!linesAndBlocks[lineIndex]) {
+        linesAndBlocks[lineIndex] = [];
+      }
+      linesAndBlocks[lineIndex].push(block);
+    }
+  });
```

Comment:
1) `map` should not be used for side effects! 😠 
2) Notice an incorrect [type](https://github.com/Ensembl/ensembl-genome-browser/blob/f912fda85a8746034d83b3fc6ffd3a2a1a930a1a/src/types.ts#L73) in `ensembl-genome-browser`: a `ZmenuContentLine` is not a `ZmenuContentBlock` — it's an array of content blocks
3) 

```
  const lines = props.feature.data.reduce((lines, block) => {
    if (block.type === 'line-break') {
      lines.push([]);
    } else {
      const lastLine = lines[lines.length-1];
      lastLine.push(block);
    }
    return lines;
  }, [[]] as ZmenuContentBlockType[][])
```

Reply:
Updated it. Thanks